engine/src/contract_methods/xcc.rs

```diff
@@ -1,17 +1,71 @@
 use crate::{
-    contract_methods::{require_owner_only, require_running, ContractError},
+    contract_methods::{predecessor_address, require_owner_only, require_running, ContractError},
+    engine::Engine,
     errors,
-    hashchain::with_hashchain,
+    hashchain::{with_hashchain, with_logs_hashchain},
     state, xcc,
 };
+use aurora_engine_modexp::AuroraModExp;
 use aurora_engine_sdk::{
     env::Env,
     io::{StorageIntermediate, IO},
     promise::PromiseHandler,
 };
-use aurora_engine_types::{borsh::BorshSerialize, types::Address};
+use aurora_engine_types::{
+    account_id::AccountId,
+    borsh::BorshSerialize,
+    format,
+    parameters::{engine::SubmitResult, xcc::WithdrawWnearToRouterArgs},
+    types::Address,
+};
 use function_name::named;
 
+#[named]
+pub fn withdraw_wnear_to_router<I: IO + Copy, E: Env, H: PromiseHandler>(
+    io: I,
+    env: &E,
+    handler: &mut H,
+) -> Result<SubmitResult, ContractError> {
+    with_logs_hashchain(io, env, function_name!(), |io| {
+        let state = state::get_state(&io)?;
+        require_running(&state)?;
+        env.assert_private_call()?;
+        let check_promise: Result<(), &[u8]> = match handler.promise_result_check() {
+            Some(true) | None => Ok(()),
+            Some(false) => Err(b"ERR_CALLBACK_OF_FAILED_PROMISE"),
+        };
+        check_promise?;
+        let args: WithdrawWnearToRouterArgs = io.read_input_borsh()?;
+        let current_account_id = env.current_account_id();
+        let recipient = AccountId::new(&format!(
+            "{}.{}",
+            args.target.encode(),
+            current_account_id.as_ref()
+        ))?;
```

Comment:
Yes, this is a known limitation of XCC. For `aurora` the account id is certainly short enough. For silos it could be a longer account id, but silos will probably also have less use for XCC. If it becomes an issue for silos then we can explore solutions at that time.

This part of the code should not be reachable if the account id is too long. The XCC precompile itself should fail. I've added a check there in https://github.com/aurora-is-near/aurora-engine/pull/864/commits/d1ca7864665f811881392b21691288bb77f6983d